test/nerves_network_interface_test.exs

```diff
@@ -0,0 +1,51 @@
+defmodule Nerves.NetworkInterfaceTest do
+  use ExUnit.Case, async: false
+
+  @moduletag :external
+  
+  defp add_dev(name) do
+    {_, 0} = System.cmd("sudo",  ~w"ip link add #{name}0 type veth peer name #{name}1")
```

Comment:
I like the idea. It can probably be cleaned up over time, but it's way better than the current situation. Can you split this PR between the tests and specs parts? I think that unless there's a snafu with CI, we should get the tests in and then iterate a couple more times on the specs.